src/PowerShellEditorServices/Services/PowerShell/Host/PsesInternalHost.cs

```diff
@@ -635,6 +769,15 @@ private void DoOneRepl(CancellationToken cancellationToken)
             {
                 // Do nothing, since we were just cancelled
             }
+            // Propagate exceptions thrown from the debugger when quitting.
+            catch (TerminateException)
+            {
+                throw;
+            }
+            catch (FlowControlException)
```

Comment:
This our own exception or a PowerShell thing or a .NET thing?

Reply:
It's a PowerShell thing. Any `break` or `continue` statement throws a flow control exception if it's not a loop. In PowerShell script it's automatically rethrown until it hits a caller who is in a loop, or until the entire stack is unwound.